src/hazelcore/Statics_Exp.re

```diff
@@ -304,9 +326,17 @@ and ana_operand =
   | Inj(NotInHole, side, body) =>
     let* (ty1, ty2) = HTyp.matched_sum(ty);
     ana(ctx, body, InjSide.pick(side, ty1, ty2));
-  | Case(StandardErrStatus(NotInHole), scrut, rules) =>
-    let* ty1 = syn(ctx, scrut);
-    ana_rules(ctx, rules, ty1, ty);
+  | Case(_ /*CaseNotInHole*/, scrut, rules) =>
+    let* ty_scrut = syn(ctx, scrut);
+    ana_rules(ctx, rules, ty_scrut, ty);
+  // TODO(andrew): this triggers MissingCursorInfo when adding a concrete type
```

Comment:
I think we do want to check even in the ana case whether InconsistentBranches errors are accurate so:

* If it is a Syn error then treat it as a synthetic case and do a subsumption (don't think this will ever actually happen in the callers to ana, but correctness wise)
* If it is an Ana error, then make sure that the rules synthesize types but don't need to do a join, as in the syn case